File: game/scripts/vscripts/scenarios/default-game.lua
local ____lualib = require("lualib_bundle")
local __TS__Class = ____lualib.__TS__Class
local __TS__SourceMapTraceBack = ____lualib.__TS__SourceMapTraceBack
__TS__SourceMapTraceBack(debug.getinfo(1).short_src, {["7"] = 12,["8"] = 12,["9"] = 12,["10"] = 14,["11"] = 14,["12"] = 15,["13"] = 16,["14"] = 13,["15"] = 22,["16"] = 23,["17"] = 26,["18"] = 27,["19"] = 30,["20"] = 33,["21"] = 36,["22"] = 39,["23"] = 39,["24"] = 39,["25"] = 40,["26"] = 41,["27"] = 39,["28"] = 39,["29"] = 22});
local ____exports = {}
--- Основной режим игры
____exports.DefaultGameScenario = __TS__Class()
local DefaultGameScenario = ____exports.DefaultGameScenario
DefaultGameScenario.name = "DefaultGameScenario"
function DefaultGameScenario.prototype.____constructor(self, stageManager, playerManager, peaceMode)
    self.stageManager = stageManager
    self.playerManager = playerManager
    self.peaceMode = peaceMode
end
function DefaultGameScenario.prototype.start(self)
    print("=== Starting DEFAULT GAME scenario ===")
    self.peaceMode:enable()
    self.peaceMode:applyToAll()
    self.playerManager:giveGoldToAll(2000)
    self.playerManager:disableUnitSharing()
    self.playerManager:saveCurrentHeroes()
    Timers:CreateTimer(
        0.5,
        function()
            self.stageManager:startHeroSelection()
            return nil
        end
    )
end
return ____exports

File: src/vscripts/scenarios/default-game.ts
/**
 * Основной игровой сценарий (Custom Hero Chaos)
 */

import { StageManager } from "../game-stages/stage-manager";
import { PlayerManager } from "../players/player-manager";
import { PeaceMode } from "../combat/peace-mode";

/**
 * Основной режим игры
 */
export class DefaultGameScenario {
    constructor(
        private stageManager: StageManager,
        private playerManager: PlayerManager,
        private peaceMode: PeaceMode
    ) {}

    /**
     * Запустить основной игровой режим
     */
    start(): void {
        print("=== Starting DEFAULT GAME scenario ===");
        
        // Флоу:
        // - HERO_SELECTION + STRATEGY_TIME обрабатывает движок (кастомный пик + аспекты)
        // - тут мы стартуем только когда игра уже началась (GAME_IN_PROGRESS): PRE_COMBAT 5с -> COMBAT
        this.peaceMode.enable();
        this.peaceMode.applyToAll();
        
        // Золото игрокам
        this.playerManager.giveGoldToAll(2000);
        
        // Запрет шэринга
        this.playerManager.disableUnitSharing();
        
        // Сохраняем героев
        this.playerManager.saveCurrentHeroes();
        
        // Запускаем стадии: сразу PRE_COMBAT на 5 сек, затем первый раунд
        Timers.CreateTimer(0.5, () => {
            this.stageManager.startPreCombat(5);
            return undefined;
        });
    }
}

